### PR TITLE
Fix `@retry`, `@serial` and `@allow.skipped` tags inheritance

### PR DESCRIPTION
**Pull request by ilslv**
## Synopsis

Currently only `Scenario`s with `@retry`, `@serial` and `@allow.skipped` are affected, tags from `Feature`s and `Rule`s aren't propagated. 

## Solution

Inherit tags, as described in documentation. 

## Checklist

- Created PR:
    - [x] In [draft mode][l:1]
    - [x] Name contains `Draft: ` prefix
    - [x] Name contains issue reference
    - [x] Has assignee
- [x] Documentation is updated (if required)
- [x] Tests are updated (if required)
- [x] Changes conform code style
- [x] CHANGELOG entry is added (if required)
- [x] FCM (final commit message) is posted
    - [x] and approved
- [x] [Review][l:2] is completed and changes are approved
- Before merge:
    - [x] Milestone is set
    - [x] PR's name and description are correct and up-to-date
    - [x] `Draft: ` prefix is removed
    - [x] All temporary labels are removed

[l:1]: https://help.github.com/en/articles/about-pull-requests#draft-pull-requests
[l:2]: https://help.github.com/en/articles/reviewing-changes-in-pull-requests
[format]: https://github.com/rust-lang/rust/issues/49359

### CODE REVIEW

**Comment by ilslv** [comment]
## FCM

```
Fix `@retry`, `@serial` and `@allow.skipped` tags inheritance (#237)
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,15 +6,22 @@ All user visible changes to `cucumber` crate will be documented in this file. Th
...
-[0.15.2]: /../../tree/v0.15.2
+## [0.16.0] · 2022-10-??
+[0.16.0]: /../../tree/v0.16.0
 
-[Diff](/../../compare/v0.15.1...v0.15.2) | [Milestone](/../../milestone/17)
+[Diff](/../../compare/v0.15.1...v0.16.0) | [Milestone](/../../milestone/17)
+
+### BC Breaks
+
+- Upgraded [`gherkin`] crate to 0.13 version. ([4cad49f])
```
Comment (tyranron) [inline CHANGELOG.md:16]: @ilslv we don't require `cucumber` users to use `gherkin` crate separately. If the user uses `cucumber` only, it's not a breaking change him.

Comment (ilslv) [reply CHANGELOG.md:16]: @tyranron the same logic could be applied to upgrading `clap` or even previous versions bump of `gherkin`. We've considered all of them as a breaking changes.

Comment (tyranron) [reply CHANGELOG.md:16]: @ilslv nope. Previous versions of `gherkin` had either API changes, or at that time `gherkin` was required to be imported separately (there was no re-export and examples pointed to do so).

Regarding the `clap` crate, the reasons are different:
1. Users do use `clap` in their code in conjunction with `cucumber`, not only import it from `cucumber` (we re-export too little). So a new major version of `clap` will break that code.
2. `clap` major version updates were really breaking for the APIs.

In this case, the new `gherkin` release is major only due to MSRV bump. No API changes there. And we expect users just to use `gherkin` types we expose in `cucumber`. I don't think that some custom `gherkin`-based machinery, which a user has and wants it to integrate with APIs exposed by `cucumber` is a real case here.

**Review by tyranron — APPROVED** [review]

**Pull request merged**